assets/agenda/components/AgendaList.tsx

```diff
@@ -414,6 +414,7 @@ class AgendaList extends React.Component<IProps, IState> {
     }
 
     render() {
+        const lastGroupWithItems = this.props.groupedItems.findLastIndex((group) => group.items.length > 0);
```

Comment:
We have the [groupItems](https://github.com/superdesk/newsroom-core/blob/develop/assets/agenda/utils.ts#L699) function in agenda/utils.ts that provides the logic around grouping.

Moving this filtering code to that function would help by having a single source of truth, if the group is in the store then it should be rendered.

Or is there another reason to filter groups in the view separately to the store?

Reply:
I'm not sure if it should be there, the items still belong to that group we just don't want to see the group yet, so imo it's rather for the rendering